src/View/ViewBuilder.php

```diff
@@ -0,0 +1,332 @@
+     * Get/set the name of the view file to render. The name specified is the
+     * filename in /app/Template/<SubFolder> without the .ctp extension.
+     *
+     * @param string|null $name View file name to set. If null returns current name.
+     * @return string|$this
+     */
+    public function template($name = null)
+    {
+        if ($name === null) {
+            return $this->template;
+        }
+
+        $this->template = $name;
+        return $this;
+    }
+
+    /**
+     * Get/set the name of the layout file to render the view inside of.
+     * The name specified is the filename of the layout in /app/Template/Layout
+     * without the .ctp extension.
+     *
+     * @param string|null $name Layout file name to set. If null returns current name.
+     * @return string|$this
+     */
+    public function layout($name = null)
+    {
+        if ($name === null) {
+            return $this->layout;
+        }
+
+        $this->layout = $name;
+        return $this;
+    }
+
+    /**
+     * Set additional options for the view.
+     *
+     * @param array|null $options Either an array of options or null to get current options.
+     * @param bool $merge Whether or not to merge existing data with the new data.
+     * @return array|$this
+     */
+    public function options(array $options = null, $merge = true)
+    {
+        if ($options === null) {
+            return $this->options;
+        }
+        if ($merge) {
+            $options = array_merge($this->options, $options);
+        }
+        $this->options = $options;
+        return $this;
+    }
+
+    /**
+     * Get/set the view name
+     *
+     * @param string|null $name The name of the view
+     * @return array|$this
+     */
+    public function name($name = null)
+    {
+        if ($name === null) {
+            return $this->name;
+        }
+        $this->name = $name;
+        return $this;
+    }
+
+    /**
+     * Get/set the view classname
+     *
+     * @param string|null $name The class name for the view. Can
+     *   be a plugin.class name reference, a short alias, or a fully
+     *   namespaced name.
+     * @return array|$this
+     */
+    public function className($name = null)
+    {
+        if ($name === null) {
+            return $this->className;
+        }
+        $this->className = $name;
+        return $this;
+    }
+
+    /**
+     * Using the data in the builder, create a view instance.
+     *
+     * @param array $vars The view variables/context to use.
+     * @param \Cake\Network\Request $request The request to use.
+     * @param \Cake\Network\Response $response The response to use.
+     * @param \Cake\Event\EventManager $events The event manager to use.
+     * @return \Cake\View\View
+     * @throws \Cake\View\Exception\MissingViewException
+     */
+    public function build($vars = [], Request $request = null, Response $response = null, EventManager $events = null)
+    {
+        if ($this->className === 'View') {
```

Comment:
Not sure if this is the good place but something like
```php
if ($this->className === null) {
            $this->className = App::className('App', 'View', 'View');
            if ($this->className === false) {
                $this->className = 'Cake\View\View';
            }
        }
```
or smater should be added because I might be wrong but right now the AppView that as automatically used is not.